File: app/src/main/java/com/cere/csplayer/fragment/BaseFragment.java
package com.cere.csplayer.fragment;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

/**
 * Created by CheRevir on 2019/10/26
 */
public class BaseFragment extends Fragment {

    public static BaseFragment newInstance() {
        Bundle bundle = new Bundle();
        BaseFragment fragment = new BaseFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }
}

File: app/src/main/java/com/cere/csplayer/ui/BaseFragment.kt
package com.cere.csplayer.ui

import android.content.Context
import androidx.fragment.app.Fragment

/**
 * Created by CheRevir on 2020/9/12
 */
abstract class BaseFragment : Fragment() {


    override fun onAttach(context: Context) {
        super.onAttach(context)
        (context as OnFragmentChangeListener).onFragmentChange(this)
    }

    open fun onBackPressed(): Boolean {
        return false
    }

    open fun onAction(string: String) {

    }

    interface OnFragmentChangeListener {
        fun onFragmentChange(fragment: BaseFragment)
    }
}